GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/IncludedFolderTests.cs

```diff
@@ -0,0 +1,262 @@
+            folder.ShouldNotExistOnDisk(this.fileSystem);
+            folder = this.Enlistment.GetVirtualPathTo("GVFS", "GVFS.Common");
+            folder.ShouldNotExistOnDisk(this.fileSystem);
+        }
+
+        [TestCase, Order(2)]
+        public void AddingParentDirectoryShouldMakeItRecursive()
+        {
+            string childPath = Path.Combine(this.mainIncludedFolder, "CommandLine");
+            this.gvfsProcess.AddIncludedFolders(childPath);
+            string[] directories = Directory.GetDirectories(Path.Combine(this.Enlistment.RepoRoot, this.mainIncludedFolder));
+            directories.Length.ShouldEqual(1);
+            directories[0].ShouldEqual(Path.Combine(this.Enlistment.RepoRoot, childPath));
+            this.ValidateIncludedFolders(childPath);
+
+            this.gvfsProcess.AddIncludedFolders(this.mainIncludedFolder);
+            directories = Directory.GetDirectories(Path.Combine(this.Enlistment.RepoRoot, this.mainIncludedFolder));
+            directories.ShouldMatchInOrder(this.directoriesInMainFolder);
+            this.ValidateIncludedFolders(childPath, this.mainIncludedFolder);
+        }
+
+        [TestCase, Order(3)]
+        public void AddingSiblingFolderShouldNotMakeParentRecursive()
+        {
+            this.gvfsProcess.AddIncludedFolders(this.mainIncludedFolder);
+            this.ValidateIncludedFolders(this.mainIncludedFolder);
+
+            // Add and remove sibling folder to main folder
+            string siblingPath = Path.Combine("GVFS", "FastFetch");
+            this.gvfsProcess.AddIncludedFolders(siblingPath);
+            string folder = this.Enlistment.GetVirtualPathTo(siblingPath);
+            folder.ShouldBeADirectory(this.fileSystem);
+            this.ValidateIncludedFolders(this.mainIncludedFolder, siblingPath);
+
+            this.gvfsProcess.RemoveIncludedFolders(siblingPath);
+            folder.ShouldNotExistOnDisk(this.fileSystem);
+            folder = this.Enlistment.GetVirtualPathTo(this.mainIncludedFolder);
+            folder.ShouldBeADirectory(this.fileSystem);
+            this.ValidateIncludedFolders(this.mainIncludedFolder);
+        }
+
+        [TestCase, Order(4)]
+        public void AddingSubfolderShouldKeepParentRecursive()
+        {
+            this.gvfsProcess.AddIncludedFolders(this.mainIncludedFolder);
+            this.ValidateIncludedFolders(this.mainIncludedFolder);
+
+            // Add subfolder of main folder and make sure it stays recursive
+            string subFolder = Path.Combine(this.mainIncludedFolder, "Properties");
+            this.gvfsProcess.AddIncludedFolders(subFolder);
+            string folder = this.Enlistment.GetVirtualPathTo(subFolder);
+            folder.ShouldBeADirectory(this.fileSystem);
+            this.ValidateIncludedFolders(this.mainIncludedFolder, subFolder);
+
+            folder = this.Enlistment.GetVirtualPathTo(this.mainIncludedFolder, "CommandLine");
+            folder.ShouldBeADirectory(this.fileSystem);

```

Comment:
Also validate that `Path.Combine(this.mainIncludedFolder, "Properties")` is still present?